primitives/core/src/ecdsa.rs

```diff
@@ -103,8 +103,18 @@ impl Public {
 	/// This will convert the full public key into the compressed format.
 	#[cfg(feature = "std")]
 	pub fn from_full(full: &[u8]) -> Result<Self, ()> {
-		secp256k1::PublicKey::parse_slice(full, None)
-			.map(|k| k.serialize_compressed())
+		let pubkey = if full.len() == 64 {
```

Comment:
The `libsecp256k1` library is handling 3 kind of keys:
full (65 bytes), raw (64 bytes; no tag) and compressed.

Here we only handle the `raw` case, are you sure that `secp256k1` is handling both full and compressed case?

Reply:
Yes. The C lib handles 33 and 65 byte signatures. But no 64 byte signatures:
https://github.com/bitcoin-core/secp256k1/blob/9a5a87e0f1276e0284446af1172056ea4693737f/src/eckey_impl.h#L17-L35